### PR TITLE
feat(project): add identity decorator as to support duplicate resources

### PR DESCRIPTION
**Pull request by mroseboom**
At this moment it's not possible to use the same resource key for multiple endpoints. With the identifier decorator this will be possible and when no given it will fallback to the resource decorator.

### CODE REVIEW

File: doc/decorators.md [hunk 1]
```diff
@@ -42,7 +42,17 @@ class HelloWorld {}
 @resource('i-want-bacon')
 class HelloWorld {}
 ```
+## @identifier()
 
+This decorator comes in handy when you have the same resource name for multiple endpoints. Without it, aurelia-orm will use the resource decorator. The identifier will be used to register and retrieve the repository for the given entity.
+
+When left empty, the name of the class (.toLowerCase()) will be used as the resource name. This is usually fine.
```
Comment (RWOverdijk) [inline doc/decorators.md]: Maybe add a note that this is a bad idea for bundling, as the names of classes get renamed.

File: src/entity-manager.js [hunk 2]
```diff
@@ -47,7 +47,11 @@ export class EntityManager {
    * @chainable
    */
   registerEntity(EntityClass) {
-    this.entities[OrmMetadata.forTarget(EntityClass).fetch('resource')] = EntityClass;
+    let reference = (OrmMetadata.forTarget(EntityClass).fetch('identifier') !== null)
```
Comment (RWOverdijk) [inline src/entity-manager.js]: This works fine, but I thought I'd show you an alternative:

```js
let meta = OrmMetadata.forTarget(EntityClass);

 this.entities[meta.fetch('identifier', meta.fetch('resource'))] = EntityClass;
```

File: src/entity-manager.js [hunk 3]
```diff
@@ -62,12 +66,17 @@ export class EntityManager {
    */
   getRepository(entity) {
     let reference = this.resolveEntityReference(entity);
+    let identifier = entity;
```
Comment (RWOverdijk) [inline src/entity-manager.js]: Align consecutive assignments:

```js
let foo  = 'bar';
let cake = 'bat';
```

File: test/decorator/identifier.spec.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// import {WithIdentifier} from '../resources/entity/with-identifier';
+// import {OrmMetadata} from '../../src/orm-metadata';
+
+// describe('@identifier()', function() {
```
Comment (RWOverdijk) [inline test/decorator/identifier.spec.js]: Why is this commented out?

**Review by RWOverdijk — CHANGES_REQUESTED** [review]

**Comment by RWOverdijk** [comment]
👏 awesome! This is near a perfect PR. :)

I've added some minor feedback, and I have one request concerning the commit message. Could you change it to:

`feat(project): add identity decorator as to support duplicate resources`

Thank you for your contribution, it's appreciated.

File: test/decorator/identifier.spec.js (hunk 4 above)
Comment (mroseboom) [reply test/decorator/identifier.spec.js]: Oops, must be uncommented ;)

File: doc/decorators.md (hunk 1 above)
Comment (mroseboom) [reply doc/decorators.md]: Can you explain further why this have a negative effect for bundling? And is this applied when you use this decorator?

Comment (RWOverdijk) [reply doc/decorators.md]: @mroseboom It's the same for resource. When bundled, and minified, the class names get changed (to be shorter). Sometimes even when transpiled. Your identifier is a string, which will then no longer match.

**Comment by RWOverdijk** [comment]
I'll re-review this when I get home :) One thing I'd add is an additional test with actual duplicates. One that fails to satisfy the need for unique entities, and one that succeeds due to the newly added identifier. :)

**Comment by mroseboom** [comment]
Were should i place this (missing) unit tests? Within the `test/decorator/identifier.spec.js` file?

**Comment by RWOverdijk** [comment]
Exactly like that :) I'm sorry it took this long to get back to you.

**Review by RWOverdijk — APPROVED** [review]
